packages/indy-sdk-to-askar-migration/src/IndySdkToAskarMigrationUpdater.ts

```diff
@@ -0,0 +1,416 @@
+import type { AnonCredsCredentialValue } from '@aries-framework/anoncreds'
+import type { Agent, FileSystem, WalletConfig } from '@aries-framework/core'
+import type { EntryObject } from '@hyperledger/aries-askar-shared'
+
+import { AnonCredsCredentialRecord, AnonCredsLinkSecretRecord } from '@aries-framework/anoncreds'
+import { AskarWallet } from '@aries-framework/askar'
+import { InjectionSymbols, KeyDerivationMethod, JsonTransformer, TypedArrayEncoder } from '@aries-framework/core'
+import { Migration, Key, KeyAlgs, Store, StoreKeyMethod } from '@hyperledger/aries-askar-shared'
+
+import { IndySdkToAskarMigrationError } from './errors/IndySdkToAskarMigrationError'
+import { transformFromRecordTagValues } from './utils'
+
+/**
+ *
+ * Migration class to move a wallet form the indy-sdk structure to the new
+ * askar wallet structure.
+ *
+ * Right now, this is ONLY supported within React Native environments AND only sqlite.
+ *
+ * The reason it only works within React Native is that we ONLY update the
+ * keys, masterSecret and credentials for now. If you have an agent in Node.JS
+ * where it only contains these records, it may be used but we cannot
+ * guarantee a successful migration.
+ *
+ */
+export class IndySdkToAskarMigrationUpdater {
+  private store?: Store
+  private walletConfig: WalletConfig
+  private defaultLinkSecretId: string
+  private agent: Agent
+  private dbPath: string
+  private fs: FileSystem
+  private deleteOnFinish: boolean
+
+  private constructor(
+    walletConfig: WalletConfig,
+    agent: Agent,
+    dbPath: string,
+    deleteOnFinish = false,
+    defaultLinkSecretId?: string
+  ) {
+    this.walletConfig = walletConfig
+    this.dbPath = dbPath
+    this.agent = agent
+    this.fs = this.agent.dependencyManager.resolve<FileSystem>(InjectionSymbols.FileSystem)
+    this.defaultLinkSecretId = defaultLinkSecretId ?? walletConfig.id
+    this.deleteOnFinish = deleteOnFinish
+  }
+
+  public static async initialize({
+    dbPath,
+    agent,
+    deleteOnFinish,
+    defaultLinkSecretId,
+  }: {
+    dbPath: string
```

Comment:
I think most people won't really know where the database is stored. Can we make it optional and compute if not defined based on defaults?

Reply:
Not needed before merging

Reply:
I think we can, but it can be a hassle with android, iOS, *nix and Windows.

Reply:
Then it's something for the docs.